app/src/routes/api/v1/subscription.router.js

```diff
@@ -300,18 +314,48 @@ const subscriptionExists = (isForUser) => async (ctx, next) => {
     await next();
 };
 
-const isLoggedUserRequired = async (ctx, next) => {
+const checkMicroservice = (ctx) => {
     const loggedUser = SubscriptionsRouter.getUser(ctx);
+    return loggedUser.id === 'microservice';
+};
 
-    if (!loggedUser || USER_ROLES.indexOf(loggedUser.role) === -1) {
+const hasLoggedUser = (ctx) => {
+    const loggedUser = SubscriptionsRouter.getUser(ctx);
+    return loggedUser && USER_ROLES.indexOf(loggedUser.role) !== -1;
+};
+
+const checkValidLoggedUser = (ctx) => {
+    const loggedUser = SubscriptionsRouter.getUser(ctx);
+    return typeof loggedUser === 'object' && loggedUser.role;
+};
+
+const isLoggedUserRequired = async (ctx, next) => {
```

Comment:
(follow up on previous comment) ...like you did here. And, in this case, this does not return a boolean, so bad usage. This type of function, that is a "pass or crash" should be named along the lines of "ensureLoggedUser" or "validate...." or something.

Reply:
Here it was bad usage indeed. Updated the name of the middleware to `validateLoggedUserAuth` :+1: